src/test/java/build/buildfarm/worker/DequeueMatchEvaluatorTest.java

```diff
@@ -0,0 +1,300 @@
+///          varying amount of leniency when evaluating the platform
+///          properties.
+///
+@RunWith(JUnit4.class)
+public class DequeueMatchEvaluatorTest {
+
+  // Function under test: shouldKeepOperation
+  // Reason for testing: null queue entries should be kept
+  // Failure explanation: this decision has changed
+  @Test
+  public void shouldKeepOperationKeepNullQueueEntry() throws Exception {
+
+    // ARRANGE
+    DequeueMatchSettings settings = new DequeueMatchSettings();
+    SetMultimap<String, String> workerProvisions = HashMultimap.create();
+    QueueEntry entry = null;
+
+    // ACT
+    boolean shouldKeep =
+        DequeueMatchEvaluator.shouldKeepOperation(settings, workerProvisions, entry);
+
+    // ASSERT
+    assertThat(shouldKeep).isTrue();
+  }
+
+  // Function under test: shouldKeepOperation
+  // Reason for testing: empty plaform queue entries should be kept
+  // Failure explanation: properties are being evaluated differently now
+  @Test
+  public void shouldKeepOperationKeepEmptyQueueEntry() throws Exception {
+
+    // ARRANGE
+    DequeueMatchSettings settings = new DequeueMatchSettings();
+    SetMultimap<String, String> workerProvisions = HashMultimap.create();
+    QueueEntry entry = QueueEntry.newBuilder().setPlatform(Platform.newBuilder()).build();
+
+    // ACT
+    boolean shouldKeep =
+        DequeueMatchEvaluator.shouldKeepOperation(settings, workerProvisions, entry);
+
+    // ASSERT
+    assertThat(shouldKeep).isTrue();
+  }
+
+  // Function under test: shouldKeepOperation
+  // Reason for testing: the entry should be kept because the min cores are valid for the worker
+  // properties
+  // Failure explanation: either the property names changed or we evaluate these properties
+  // differently
+  @Test
+  public void shouldKeepOperationValidMinCoresQueueEntry() throws Exception {
+
+    // ARRANGE
+    DequeueMatchSettings settings = new DequeueMatchSettings();
+
+    SetMultimap<String, String> workerProvisions = HashMultimap.create();
+    workerProvisions.put("cores", "11");
+
+    QueueEntry entry =
+        QueueEntry.newBuilder()
+            .setPlatform(
+                Platform.newBuilder()
+                    .addProperties(
+                        Platform.Property.newBuilder().setName("min-cores").setValue("10")))
+            .build();
+
+    // ACT
+    boolean shouldKeep =
+        DequeueMatchEvaluator.shouldKeepOperation(settings, workerProvisions, entry);
+
+    // ASSERT
+    // the worker accepts because it has more cores than the min-cores requested
+    assertThat(shouldKeep).isTrue();
+  }
+
+  // Function under test: shouldKeepOperation
+  // Reason for testing: the entry should be kept because the min cores are valid for the worker
+  // properties
+  // Failure explanation: either the property names changed or we evaluate these properties
+  // differently
+  @Test
+  public void shouldKeepOperationValidEqualMinCoresQueueEntry() throws Exception {
```

Comment:
I feel this test can be merged with the one above `shouldKeepOperationValidMinCoresQueueEntry`, the QueueEntry settings are identical, the only difference is the worker provision settings.

I also see similar patterns in the following test functions. I feel some of them can be easily merged to make this file much shorter and possibly easier to read.

Reply:
thanks!  I was able to remove 3 similar tests and condense